tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/FindReplaceUITest.java

```diff
@@ -0,0 +1,314 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2016 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.ui.workbench.texteditor.tests;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertEquals;
+
+import java.util.Set;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Display;
+
+import org.eclipse.jface.text.Document;
+import org.eclipse.jface.text.IFindReplaceTarget;
+import org.eclipse.jface.text.TextSelection;
+import org.eclipse.jface.text.TextViewer;
+
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.internal.findandreplace.SearchOptions;
+
+/**
+ * Tests the FindReplaceDialog.
+ *
+ * @since 3.1
+ */
+public abstract class FindReplaceUITest<AccessType extends IFindReplaceUIAccess> {
+
+	@Rule
+	public TestName testName= new TestName();
+
+	protected TextViewer fTextViewer;
+
+	static void runEventQueue() {
+		Display display= PlatformUI.getWorkbench().getDisplay();
+		for (int i= 0; i < 10; i++) { // workaround for https://bugs.eclipse.org/323272
+			while (display.readAndDispatch()) {
+				// do nothing
+			}
+			try {
+				Thread.sleep(50);
+			} catch (InterruptedException e) {
+				// do nothing
+			}
+		}
+	}
+
+	protected AccessType dialog;
```

Comment:
Please avoid protected fields and provide a protected getter instead (to avoid that a subclass may override the value).

Reply:
https://github.com/eclipse-platform/eclipse.platform.ui/commit/49851d64fd5a67634b90f951a0aeddf3da74bfaa